src/common/hostProcess.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @name - Host Process
+ * @description - Describes the host process
+ * @member type - The type of the host process (electron, browser, etc)
+ * @member release - The release string of the host process
+ */
+export interface IHostProcess {
+    type: HostProcessType;
+    release: string;
+}
+
+/**
+ * @enum ELECTRON - Electron Host Process Type
+ * @enum BROWSER - Browser Host Process Type
+ */
+export enum HostProcessType {
+    Electron = 1 << 0,
+    Browser = 1 << 1,
+    All = ~(~0 << 2),
+}
+
+export function typeToFriendlyName(type: HostProcessType): string {
+    switch (type) {
+        case HostProcessType.Electron:
+            return "electron";
+        case HostProcessType.Browser:
+            return "browser";
+    }
+}
+
+import os from "os";
```

Comment:
This is not browser safe - this will likely fail in the web scenario, no?

Reply:
This is in fact browser safe, and returns the "userAgent" string.